pkg/monitoring/vms/prometheus/prometheus.go

```diff
@@ -62,7 +63,7 @@ var (
 		"kubevirt_vmi_phase_count",
 		"VMI phase.",
 		[]string{
-			"node", "phase",
+			"node", "phase", "os", "workload", "flavor",
```

Comment:
Are those VM templates the standard for creating VMs?

If I create a VM without the annotations, this will be the result:

```
kubevirt_vmi_phase_count{flavor="",node="node01",os="",phase="running",workload=""} 1
```
I ask because these labels will be created even for users who don't work with the templates, and thus, increase metric storage capacity.

Reply:
Good question, how does it work for other metrics when part of the payload isn't available ?

Reply:
For general VMI metrics such as network, storage, etc. The metric is not exposed at all if libvirtd doesn't report anything.

About dynamic labels such as the ones added to k8s metadata, we currently search and add a new label only if present.

You can see here for example: 
https://github.com/kubevirt/kubevirt/blob/131ed27576994688a71c5f4fe1b2a9356bbda7a9/pkg/monitoring/vms/prometheus/prometheus.go#L79-L101

The metric description is created inside the function that is updating it, and the labels and label values are added dynamically inside that same function. But this won't work for you I think, since this particular metric is handled differently :(

Maybe the `vmiCountMetric` type could have some extra boolean attributes: `osSet`, `flavorSet`, `workloadSet`? Similar to how we handle libvirtd metrics

Reply:
Alright, looking at another metric (kube_pod_info as an example) if some values are not present a simple <none> string will be displayed.  Pushed a patch to fix this.